pkg/controller/ground/bootstrap/csi/csi123.go

```diff
@@ -0,0 +1,217 @@
+
+	kubernikus_v1 "github.com/sapcc/kubernikus/pkg/apis/kubernikus/v1"
+	"github.com/sapcc/kubernikus/pkg/version"
+)
+
+func SeedCinderCSIPlugin123(client clientset.Interface, dynamicClient dynamic.Interface, klusterSecret *kubernikus_v1.Secret, versions version.KlusterVersion) error {
+	secret := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cloud-config",
+			Namespace: "kube-system",
+		},
+		StringData: map[string]string{
+			"cloudprovider.conf": `[Global]
+auth-url="https://identity-3.` + klusterSecret.Openstack.Region + `.cloud.sap/v3/"
+domain-name="kubernikus"
+tenant-id="` + klusterSecret.Openstack.ProjectID + `"
+username="` + klusterSecret.Openstack.Username + `"
+password="` + klusterSecret.Openstack.Password + `"
+region="` + klusterSecret.Openstack.Region + `"
+
+[BlockStorage]
+rescan-on-resize = yes
+`,
+		},
+	}
+
+	err := createSecret(client, &secret)
+	if err != nil {
+		return errors.Wrap(err, "CSISecret")
+	}
+
+	err = createServiceAccount(client, CSIServiceAccountController)
+	if err != nil {
+		return errors.Wrap(err, "CSIServiceAccountController")
+	}
+
+	err = createServiceAccount(client, CSIServiceAccountNode)
+	if err != nil {
+		return errors.Wrap(err, "CSIServiceAccountNode")
+	}
+
+	err = createRole(client, CSIRole)
+	if err != nil {
+		return errors.Wrap(err, "CSIRole")
+	}
+
+	err = createClusterRole(client, CSIClusterRoleAttacher123)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleAttacher")
+	}
+
+	err = createClusterRole(client, CSIClusterRoleNodePlugin)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleNodePlugin")
+	}
+
+	err = createClusterRole(client, CSIClusterRoleProvisioner123)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleProvisioner")
+	}
+
+	err = createClusterRole(client, CSIClusterRoleResizer123)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleResizer")
+	}
+
+	err = createClusterRole(client, CSIClusterRoleSnapshotter123)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleSnapshotter")
+	}
+
+	err = createRoleBinding(client, CSIRoleBindingResizer)
+	if err != nil {
+		return errors.Wrap(err, "CSIRoleBindingResizer")
+	}
+
+	err = createClusterRoleBinding(client, CSIClusterRoleBindingAttacher)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleBindingAttacher")
+	}
+
+	err = createClusterRoleBinding(client, CSIClusterRoleBindingNodePlugin)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleBindingNodePlugin")
+	}
+
+	err = createClusterRoleBinding(client, CSIClusterRoleBindingProvisioner)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleBindingProvisioner")
+	}
+
+	err = createClusterRoleBinding(client, CSIClusterRoleBindingResizer)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleBindingResizer")
+	}
+
+	err = createClusterRoleBinding(client, CSIClusterRoleBindingSnapshotter)
+	if err != nil {
+		return errors.Wrap(err, "CSIClusterRoleBindingSnapshotter")
+	}
```

Comment:
Can we just call `SeedCinderCSIRoles123` instead?

Reply:
Yeah, but this is only a subset of changes used on kluster upgrade. I found this more confusing.